Add documentation about GPU features (#233)
JuliaSmoothOptimizers/Krylov.jl: docs/src/gpu.md

```

Optimized operator-vector products that exploit GPU features can be also used by means of linear operators.

> Preconditioners, especially incomplete Cholesky or Incomplete LU factorizations with triangular solves,

Reviewer: ```suggestion
Preconditioners, especially incomplete Cholesky or Incomplete LU factorizations that involve triangular solves,
```

> can be performed directly on GPU with efficient operators that take advantage of CUSPARSE routines.

Reviewer: ```suggestion
can be applied directly on GPU thanks to efficient operators that take advantage of CUSPARSE routines.
```

> ```julia

Reviewer: ```suggestion
### Example with a symmetric positive-definite system

```julia
```

using CUDA, Krylov
using CUDA.CUSPARSE, SparseArrays

# LLᵀ ≈ A
P = ic02(A_gpu, 'O')

# Solve Py = x
function ldiv!(y, P, x)
  y .= x
  sv2!('T', 'L', 1.0, P, y, 'O')
  sv2!('N', 'U', 1.0, P, y, 'O')
  return y
end

# Operator that model P⁻¹
y = similar(b_gpu); n = length(b_gpu); T = eltype(b_gpu)
opM = LinearOperator(T, n, n, true, true, x -> ldiv!(y, P, x))

# Solve a symmetric positive definite system with an incomplete Cholesky preconditioner on GPU
(x, stats) = cg(A_gpu, b_gpu, M=opM)
```

> ```julia

Reviewer: ```suggestion
### Example with a general square system

```julia
```

```

Optimized operator-vector products that exploit GPU features can be also used by means of linear operators.

Preconditioners, especially incomplete Cholesky or Incomplete LU factorizations that involve triangular solves,
can be applied directly on GPU thanks to efficient operators that take advantage of CUSPARSE routines.

### Example with a symmetric positive-definite system

```julia
using CUDA, Krylov
using CUDA.CUSPARSE, SparseArrays

# LLᵀ ≈ A
P = ic02(A_gpu, 'O')

# Solve Py = x
function ldiv!(y, P, x)
  y .= x

>   sv2!('T', 'L', 1.0, P, y, 'O')

Author: sv2!('T', 'U', 1.0, P, y, 'O')
